src/main/java/demo/dao/ResultSetParser.java

```diff
@@ -0,0 +1,25 @@
+package demo.dao;
+
+import demo.annotation.Column;
+import lombok.SneakyThrows;
+import java.lang.reflect.Constructor;
+import java.lang.reflect.Field;
+import java.sql.ResultSet;
+
+public class ResultSetParser {
+    @SneakyThrows
+    public <R> R parseToEntity(Class<R> clazz, ResultSet resultSet) {
+        Constructor<?> constructor = clazz.getDeclaredConstructor();
+        Object entity = constructor.newInstance();
+        Field[] declaredFields = clazz.getDeclaredFields();
+        for (Field field : declaredFields) {
+            Column annotation = field.getAnnotation(Column.class);
```

Comment:
@romanovosad87 what if the field is not annotated with `@Column`? You cover this case in `getIdFieldName` method but not here... Generally speaking, you should have a logic that allows you to **get column name by field and it's the same for id and other columns.**